sql/expression/function/strcmp.go

```diff
@@ -92,11 +88,21 @@ func (s *StrCmp) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		return nil, nil
 	}
 
-	str2, err := sql.ConvertToString(expr2, sql.LongText)
-	if err != nil {
-		return nil, err
+	if sql.IsText(s.Left.Type()) && sql.IsText(s.Right.Type()) {
+		st1 := s.Left.Type().(sql.StringType)
+		st2 := s.Right.Type().(sql.StringType)
```

Comment:
(minor) `leftStringType` and `rightStringType` would make this block just a little easier to read.